src/main/java/com/minecolonies/coremod/entity/ai/minimal/EntityAICitizenWander.java

```diff
@@ -40,15 +40,23 @@ public EntityAICitizenWander(final EntityCitizen citizen, final double speed)
     @Override
     public boolean shouldExecute()
     {
-        if (isToOld() || checkForRandom() || citizen.getDesiredActivity() == EntityCitizen.DesiredActivity.SLEEP)
+        if (isTooOld() || checkForRandom() || citizen.getDesiredActivity() == EntityCitizen.DesiredActivity.SLEEP)
         {
             return false;
         }
 
-        Vec3d vec3d = RandomPositionGenerator.findRandomTarget(citizen, 10, 7);
-        if (vec3d == null)
+        Vec3d vec3d;
+        if(citizen.getCitizenData().getSaturation() <= 0)
         {
-            return false;
+            vec3d = new Vec3d(citizen.getColony().getBuildingManager().getBestRestaurant(citizen));
+        }
+        else
+        {
+            vec3d = RandomPositionGenerator.findRandomTarget(citizen, 10, 7);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 10 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 7 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)